src/components/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+import {blockNew} from '../utils/cn';
+import {isOfType} from '../utils/isOfType';
+
+import {DisclosureProvider} from './DisclosureContext';
+import {DisclosureDetails} from './DisclosureDetails/DisclosureDetails';
+import {DefaultDisclosureSummary, DisclosureSummary} from './DisclosureSummary/DisclosureSummary';
+
+import './Disclosure.scss';
+
+export const b = blockNew('disclosure');
+
+export type DisclosureSize = 'm' | 'l' | 'xl';
+export type DisclosureArrowPosition = 'left' | 'right';
+
+export interface DisclosureComposition {
+    Summary: typeof DisclosureSummary;
+}
+
+export interface DisclosureProps {
+    /** Disclosure size */
+    size: DisclosureSize;
+    /** Disabled state */
+    disabled?: boolean;
+    /** Default opening state */
+    defaultExpanded?: boolean;
+    /** Control position */
+    arrowPosition?: DisclosureArrowPosition;
+    /** Content summary */
+    summary?: string;
+    /** Class name */
+    className?: string;
+    /** Content */
+    children?: React.ReactNode;
+    /** Keep content in DOM */
+    keepMounted?: boolean;
+    /** Callback fired when the expand/collapse state is changed  */
+    onChange?: (event: React.SyntheticEvent, expanded: boolean) => void;
+}
+
+const isDisclosureSummaryComponent = isOfType(DisclosureSummary);
+
+// @ts-ignore this ts-error is appears when forwarding ref. It complains that DisclosureComposition props is not provided initially
+export const Disclosure: React.FunctionComponent<DisclosureProps> & DisclosureComposition =
+    React.forwardRef<HTMLDivElement, DisclosureProps>(function Disclosure(props, ref) {
+        const {
+            size = 'xl',
+            disabled = false,
+            defaultExpanded = false,
+            arrowPosition = 'left',
+            summary = '',
+            className,
+            keepMounted = true,
+            children,
+            onChange = () => {},
```

Comment:
Why this mocked function? We can write smth like `onChange?.(...)`

Reply:
I did it wilfully, because I wanted all properties in `DisclosureProvider` to be required. This will ensure that if new high-level props is added, developer will not forget to pass it to provider. And also I did't want to bloat `Omit` part in `DisclosureProviderProps`, so this mock has been added.